### PR TITLE
[DOC] Fix style issues and add videos to tracing and profiling shared files

### PR DESCRIPTION
**Pull request by knylander-grafana**
**Why do we need this feature?**

Moves images from the /static to /media GCP areas for screenshots used in tracing shared files and the Pyroscope data source page. Add recently released YouTube videos to tracing and profiling docs. 

@Eve832 @jdbaldry This PR addresses the image issues and style issues you found in these pages. 

@Rperry2174 I added traces to profiles to the Pyroscope data source page and updated the formatting for the query editor. 

Fixes: https://github.com/grafana/tempo-squad/issues/375

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/datasources/grafana-pyroscope.md
Comment (knylander-grafana) [inline docs/sources/datasources/grafana-pyroscope.md:45]: @Rperry2174 Do we need to do anything with the two sentences that refer to Phlare on line 45? Should they be updated in ay way or removed?

**Review by jdbaldry — COMMENTED** [review]
Thanks for putting this together!

File: docs/sources/shared/datasources/tempo-traces-to-profiles.md
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-traces-to-profiles.md:59]: Thank you for removing future tense. Slight wording suggestion to tighten the text a little: 
```suggestion
   These tags can be used in the custom query with `${__tags}` variable. This variable interpolates the mapped tags as list in an appropriate syntax for the data source. Only the tags that were present in the span are included; tags that aren't present are omitted. You can also configure a new name for the tag. This is useful in cases where the tag has dots in the name and the target data source doesn't allow using dots in labels. For example, you can remap `service.name` to `service_name`. If you don’t map any tags here, you can still use any tag in the query, for example: `method="${__span.tags.method}"`.
```

File: docs/sources/datasources/grafana-pyroscope.md
Comment (jdbaldry) [inline docs/sources/datasources/grafana-pyroscope.md:86]: The spacing in this section looks off.

There are two spaces after each list item (which seems to render fine but is unusual).

File: docs/sources/datasources/grafana-pyroscope.md
Comment (jdbaldry) [inline docs/sources/datasources/grafana-pyroscope.md:79]: With this level of indentation, this section is rendered as a code block.

File: docs/sources/datasources/tempo/_index.md
Comment (jdbaldry) [inline docs/sources/datasources/tempo/_index.md:31]: Same question as in https://github.com/grafana/website/pull/17547/files#r1466714511

File: docs/sources/datasources/tempo/configure-tempo-data-source.md
Comment (jdbaldry) [inline docs/sources/datasources/tempo/configure-tempo-data-source.md:42]: Same question as in https://github.com/grafana/website/pull/17547/files#r1466714511

File: docs/sources/datasources/grafana-pyroscope.md
Comment (knylander-grafana) [reply docs/sources/datasources/grafana-pyroscope.md:86]: I'm not sure why. I'll check the formatting and fix it. Thank you.

File: docs/sources/datasources/grafana-pyroscope.md
Comment (knylander-grafana) [reply docs/sources/datasources/grafana-pyroscope.md:79]: it's weird because it looked okay in the preview and didn't appear as a code block. I should try it again and verify. 
Happy to fix it.

Comment (jdbaldry) [reply docs/sources/datasources/grafana-pyroscope.md:79]: Screenshot 2024-01-25 at 21 24 11

Probably changed as a result of running `prettier` when the file was in an ambiguous form. 

I'd expect the indentation of the paragraphs to match the indentation of the first character of the list item.

For example:

```markdown
1. List item.

   Paragraph as part of the first list item.
```

File: docs/sources/datasources/grafana-pyroscope.md
Comment (Rperry2174) [reply docs/sources/datasources/grafana-pyroscope.md:45]: yeah we should replace phlare with pyroscope everywhere here (and remove any remaining references to phlare

File: docs/sources/datasources/grafana-pyroscope.md
Comment (Rperry2174) [inline docs/sources/datasources/grafana-pyroscope.md:24]:
```suggestion
Grafana Pyroscope is a horizontally scalable, highly available, multi-tenant, OSS, continuous profiling aggregation system. Add it as a data source, and you are ready to query your profiles in [Explore][explore].
```

File: docs/sources/datasources/grafana-pyroscope.md
Comment (Rperry2174) [inline docs/sources/datasources/grafana-pyroscope.md:41]:
```suggestion
   | `URL`          | The URL of the Grafana Pyroscope instance, e.g., `http://localhost:4100`                                                                                                                                                                                                                                        |
```

File: docs/sources/datasources/grafana-pyroscope.md
Comment (Rperry2174) [inline docs/sources/datasources/grafana-pyroscope.md:86]: is there supposed to be a link here?

Comment (jdbaldry) [reply docs/sources/datasources/grafana-pyroscope.md:86]: This is a less common version of Markdown link known as [_reference-style_](https://spec.commonmark.org/0.30/#link-reference-definitions).

The destination is defined in https://github.com/grafana/grafana/pull/81215/files/4eb82167a07ec87480c6330fd32801203de56679#diff-74748b5f590531c97ca006dcf25af41006921b78039b7fe41754e62f3cd66a3eR129-R130.

File: docs/sources/datasources/tempo/_index.md
Comment (knylander-grafana) [reply docs/sources/datasources/tempo/_index.md:31]: I'll add the reference here and try the video time.

Comment (knylander-grafana) [reply docs/sources/datasources/tempo/_index.md]: I tried this in the Grafana docs in the other PR by adding the line below and it had no effect. I'm not going to try and add the time stamp here. This is the line I used: 
`{{< youtube id="cqHO0oYW6Ic" start="298" >}}`

File: docs/sources/datasources/tempo/configure-tempo-data-source.md
Comment (knylander-grafana) [reply docs/sources/datasources/tempo/configure-tempo-data-source.md]: I tried adding the time start and it didn't have an impact, so for now I'm ont going to hold up this PR to try and troubleshoot it.

File: docs/sources/datasources/grafana-pyroscope.md
Comment (knylander-grafana) [reply docs/sources/datasources/grafana-pyroscope.md:45]: I searched through the Pyroscope page and didn't find any more references to Phlare.

File: docs/sources/datasources/grafana-pyroscope.md
Comment (knylander-grafana) [reply docs/sources/datasources/grafana-pyroscope.md]: I think you're right that it was Prettier that added the spaces. They should be fixed now. I removed the extra spaces and put things back in line.

File: docs/sources/datasources/tempo/_index.md
Comment (knylander-grafana) [inline docs/sources/datasources/tempo/_index.md:32]: @jdbaldry I tried this but it didn't seem to work on the local build. I'm not going to worry about it. We have the note in the text on where to start. We can figure it out later and iterate on it.

File: docs/sources/shared/datasources/tempo-editor-traceql.md
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-editor-traceql.md:61]: This line has a conflict in the PR. If you try to resolve the conflicts, use the content from the version in this PR.

File: docs/sources/shared/datasources/tempo-traces-to-profiles.md
Comment (knylander-grafana) [inline docs/sources/shared/datasources/tempo-traces-to-profiles.md:39]: Lines 33-39 in this file resolve the conflict from the PR. I incorporated the content that was missing from this file.

**Comment by knylander-grafana** [comment]
Reviewer -- I don't have merge rights so I can't resolve the conflicts in the PR. However, I've been through the conflicts and updated the file accordingly. You should be good to accept the changes present in the PR files. If you find something different, please ping me and I"ll be happy to help.

File: docs/sources/datasources/grafana-pyroscope.md
Comment (clayton-cornell) [inline docs/sources/datasources/grafana-pyroscope.md:41]:
```suggestion
   | `URL`          | The URL of the Grafana Pyroscope instance, for example, `http://localhost:4100`.                                                                                                                                                                                                                                       |
```

Or does it add confusion if there's a full-stop here?

File: docs/sources/shared/datasources/tempo-traces-to-profiles.md
Comment (clayton-cornell) [inline docs/sources/shared/datasources/tempo-traces-to-profiles.md:48]: I don't know the terminology here... is `a trace to profiles span link` a name for the kind of link? or is a word missing here?

**Review by clayton-cornell — COMMENTED** [review]
LGTM. Two minor points.

**Review by clayton-cornell — APPROVED** [review]

File: docs/sources/datasources/grafana-pyroscope.md
Comment (knylander-grafana) [reply docs/sources/datasources/grafana-pyroscope.md:41]: Nope. The other rows in the tables have periods, so I think that is good

File: docs/sources/shared/datasources/tempo-traces-to-profiles.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-traces-to-profiles.md:48]: So the feature name is "trace to profiles", which makes me wonder if it should be "traces-to-profiles span link" since traces-to-spans is functioning as an adjective.

Comment (clayton-cornell) [reply docs/sources/shared/datasources/tempo-traces-to-profiles.md:48]: The hyphens def help :-)

File: docs/sources/datasources/grafana-pyroscope.md
Comment (knylander-grafana) [reply docs/sources/datasources/grafana-pyroscope.md]: It should be good now.

File: docs/sources/shared/datasources/tempo-traces-to-profiles.md
Comment (knylander-grafana) [reply docs/sources/shared/datasources/tempo-traces-to-profiles.md:48]: Okay!
```suggestion
   The tags you configure must be present in the spans attributes or resources for a trace-to-profiles span link to appear.
```

**Comment by clayton-cornell** [comment]
@knylander-grafana Fixed the conflicts

**Comment by knylander-grafana** [comment]
@clayton-cornell Thank you for the review!  If you are able to merge, would you be willing to merge this PR? I don't have merge rights.

File: docs/sources/datasources/tempo/_index.md
Comment (jdbaldry) [reply docs/sources/datasources/tempo/_index.md:32]: Bah, sorry. This will work on the website. I need to release a new version of the `grafana/docs-base` image.

**Pull request merged**
